Fix resubmit for cached csrf token by using esi for cacheable requests only (#286)
sulu/SuluFormBundle: Resources/doc/csrf.md

## Ajax

> > This solution is required when using `Varnish`:

Reviewer: ```suggestion
> This solution is required when pages are cached using `Varnish`:
```

```yaml
# config/routes/sulu_form.yaml

sulu_form.token:
    path: /_form/token
    defaults:
        _controller: Sulu\Bundle\FormBundle\Controller\FormTokenController::tokenAction
        _requestAnalyzer: false
```

### A. Ajax with jquery

> A simplified version loading the csrf token over ajax could look like this:

Reviewer: ```suggestion
A simple example for loading the csrf token over ajax looks like this:
```

```

```js
jQuery.get('/_form/token?form=' + formName + '&html=0').done(function(data) {
    jQuery('#' + fieldId).val(data);
});
```

> ### B. Ajax with sulu web js

Reviewer: ```suggestion
### B. Ajax with sulu web-js
```

> When using [`@sulu/web`](https://github.com/sulu/web-js) / [`sulu/web-twig`](https://github.com/sulu/web-twig) component library this could look like the following:

Reviewer: ```suggestion
When using [`@sulu/web`](https://github.com/sulu/web-js) / [`sulu/web-twig`](https://github.com/sulu/web-twig) component library, loading the csrf token over ajax looks like this:
```

```twig
{# templates/form/your-theme.html.twig #}

{% extends '@SuluForm/themes/basic.html.twig' %}

{%- block csrf_token_widget -%}
    {{ block('hidden_widget') }}

>     {% do register_component('csrf-token', { id: id, fornName: form.parent.vars.name }) %}

Reviewer 2: ```suggestion
    {% do register_component('csrf-token', { id: id, formName: form.parent.vars.name }) %}
```
